src/table.js

```diff
@@ -538,15 +543,99 @@ export function getTypeValidator(colType) {
   }
 }
 
+export function coerceToType(value, type, options = {}) {
+  const defaultValue = options.soft ? value : null;
+  const numberDefault = defaultValue === null ? NaN : defaultValue;
+  switch (type) {
+    case "string":
+      return value === "string"
+        ? value.trim()
```

Comment:
Applying the `.trim()` operation on strings feels like a nice feature, but I'm just pausing to think if it's expected (it's already done in [d3.autoType](https://github.com/d3/d3-dsv/blob/6d96d6c0c70add67ec3869a22b3bbcd072ce328b/src/autoType.js#L3
), so not worth discussing at length).

As a user, is there ever a case when I might want to retain the whitespace in my strings (perhaps to identify rows with data entry errors...)? I probably would always want to remove them, but that feels like a data cleaning step that's being done silently. 

If we move forward with an option to do soft-coercion, perhaps that shouldn't trim....?

Reply:
Thanks for all the valuable feedback, @mkfreeman! I've made revisions to fix almost everything you mentioned, and am going to think about the `getAllKeys` question. I'll also be making the revisions discussed in the team meeting, including removing soft coercion.